design/ceph/object/ceph-sse-s3.md

```diff
@@ -0,0 +1,44 @@
+---
+# AWS server side encryption SSE-S3 support for RGW
+target-version: release-1.10
+---
+
+# Feature Name
+AWS server side encryption SSE-S3 support for RGW
+
+## Summary
+The S3 protocol supports three different types of [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html): SSE-C, SSE-KMS and SSE-S3. For the last two RGW server need to configure with external services such as [vault](https://www.vaultproject.io/). Currently Rook configure RGW with `SSE-KMS` options to handle the S3 requests with the `sse:kms` header. Recently the support for handling the `sse:s3` was added to RGW, so Rook will now provide the option to configure RGW with `sse:s3`.
+
+The `sse:s3` is supported only from Ceph v17 an onwards, so this feature can only be enabled for Quincy or newer.
+
+### Goals
+Configure RGW with `SSE-S3` options, so that RGW can handle request with `sse:s3` headers.
+
+## Proposal details
+Introducing new field `s3` in `SecuritySpec` which defines `AWS-SSE-S3` support for RGW.
+```
+security:
+  kms:
+    ..
+  s3:
+    connectionDetails:
+      KMS_PROVIDER: vault
```

Comment:
Besides following the pattern that we already have for settings under `security.kms`, is there a reason we don't follow more standard naming convention with camel case? For example, what about:
```
security:
  s3:
    connectionDetails:
      kmsProvider: vault
      vaultAddress: ...
...
```

Consistency with `security.kms` probably makes more sense like you have it, but I want to be clear why the settings are named this way.

Reply:
@travisn : `ConnectionDetails` is [string][string] map and `security.kms` has similar values, I have copied it from https://rook.io/docs/rook/v1.9/Storage-Configuration/Advanced/key-management-system/#token-based-authentication only

Reply:
Agreed, the string map makes sense, and also some consistency with the kms settings. The all caps settings are just not standard for CRDs.